migrate.go

```diff
@@ -740,11 +739,26 @@ func (m *Migrate) runMigrations(ret <-chan interface{}) error {
 			if err := m.databaseDrv.SetVersion(migr.TargetVersion, true); err != nil {
 				return err
 			}
+			fmt.Printf("Migration Version: %v\nTargetVersion: %v", migr.Version, migr.TargetVersion)
+			migr.Down = RegisteredFunctions[int64(migr.Version)].Down
+			migr.Up = RegisteredFunctions[int64(migr.Version)].Up
+			if migr.Up != nil {
+				m.logVerbosePrintf("Read and execute migration\n")
+				if migr.TargetVersion >= int(migr.Version) {
+					fmt.Printf("Migrating Up")
+					migr.Up(m.databaseDrv)
+				} else {
+					fmt.Printf("Migrating Down")
+					migr.Down(m.databaseDrv)
+				}
 
-			if migr.Body != nil {
-				m.logVerbosePrintf("Read and execute %v\n", migr.LogString())
-				if err := m.databaseDrv.Run(migr.BufferedBody); err != nil {
-					return err
+			} else {
+				fmt.Println("Up not registered")
+				if migr.Body != nil {
+					m.logVerbosePrintf("Read and execute %v\n", migr.LogString())
+					if err := m.databaseDrv.Run(migr.BufferedBody); err != nil {
```

Comment:
@AmanKumar2696 Are we not running migr.Down() here, since the migr.UP() is empty

Reply:
Yes @seswarrajan . Both `Up()` & `Down()` migrations need to be registered.